### PR TITLE
Fix PCL Compiler Warnings on Ubuntu 14.04

### PR DESCRIPTION
**Pull request by Jebediah**
This is to address issues #39 and #40. The vtk compiler warnings are because the version of pcl in the ubuntu repositories is 1.7.2 and looks for vtk 5.8 while the version of vtk in the ubuntu repositories is a mix of 5.8, 5.10, and 6. The components being complained about don't seem to be available in version 5.8. Fortunately vtk stands for visual toolkit, and we aren't using these features so these warnings are safe to ignore.

The pcl install script now changes for trusty or xenial.

Finally another warning about virtual destructors was resolved.

### CODE REVIEW

**Comment by leokoppel** [comment]
> Fortunately vtk stands for visual toolkit, and we aren't using these features so these warnings are safe to ignore.

So this PR does not actually remove the warnings? (I see travis is still printing them)

Also, why do we install `libvtk`-related packages if we don't use them?

**Comment by Jebediah** [comment]
Yes it doesn't remove them.
vtk is a PCL dependency.

**Comment by leokoppel** [comment]
@Jebediah 

>  Yes it doesn't remove them.

This PR is called "Fix PCL Compiler Warnings on Ubuntu 14.04" so you can understand the confusion - what does this PR actually do then?

> vtk is a PCL dependency.

It is a dependency only of the `visualization` module.

File: wave_matching/src/icp.cpp [hunk 1]
```diff
@@ -31,6 +31,18 @@ ICPMatcher::ICPMatcher(float res) {
...
+    if (this->ref) {
+        this->ref.reset();
+    }
+    if (this->target) {
+        this->target.reset();
+    }
+    if (this->final) {
+        this->final.reset();
+    }
+}
```
Comment (chutsu) [inline wave_matching/src/icp.cpp]: Are these resets actally needed? won't they reset themselves when they go out of scope? Just a question.

**Comment by chutsu** [comment]
The title is my bad it was originally "Fix 39".

File: wave_matching/src/icp.cpp (hunk 1 above)
Comment (Jebediah) [inline wave_matching/src/icp.cpp]: wasn't sure. decided to play it safe

**Pull request merged**